src/Injector/Injector.ts

```diff
@@ -305,7 +305,30 @@ export class Injector {
    * Overloads allow arrays or maps to be returned.
    * @param name - Dependency
    */
-  public get = (dependencies: IDependencies) => {
+  public get<K extends keyof T>(dependencies: K): T[K];
```

Comment:
Maybe add a reference to the issue in the typescript repo so that we can track if it's been fixed or not easier?

Reply:
E.g https://github.com/Microsoft/TypeScript/issues/15972
Dunno if we should be tracking others